examples/main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
```

Comment:
Please consider using [PCG](https://pkg.go.dev/math/rand/v2#PCG) instead of a real RNG. It's good enough for our purposes of course, and the main reason it's useful is that it can be seeded, e.g. with a constant value. That means you can rerun it after you make some changes and still get the same results.

Reply:
While I agree that deterministic results have a benefit:
* math/rand shouldn't be used in crypto. 
* math/rand/v2 doesn't implement io.Reader for exactly that reason (see https://go.dev/blog/randv2#problem.read)
* I would worry that this code gets used as reference and insecure code gets copied
* Instead, I would just store the keys in the directory directly (we have private keys in the doc already)

Let's merge this PR and discuss at our next sync?